src/main/java/cloud/fogbow/ras/core/plugins/interoperability/aws/AwsV2CloudUtil.java

```diff
@@ -0,0 +1,165 @@
+import software.amazon.awssdk.services.ec2.model.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AwsV2CloudUtil {
+
+    public static final Logger LOGGER = Logger.getLogger(AwsV2CloudUtil.class);
+    public static final Object SECURITY_GROUP_RESOURCE = "Security Groups";
+    public static final String AWS_TAG_GROUP_ID = "groupId";
+    public static final String TCP_PROTOCOL = "tcp";
+    public static final String AWS_TAG_NAME = "Name";
+    public static final Integer SSH_DEFAULT_PORT = 22;
+    
+    public static Image getImagesFrom(DescribeImagesResponse response) throws FogbowException {
+        if (response != null && !response.images().isEmpty()) {
+            return response.images().listIterator().next();
+        }
+        throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
+    }
+    
+    public static DescribeImagesResponse doDescribeImagesRequest(DescribeImagesRequest request, Ec2Client client)
+            throws FogbowException {
+        try {
+            return client.describeImages(request);
+        } catch (Exception e) {
+            throw new UnexpectedException(String.format(Messages.Exception.GENERIC_EXCEPTION, e), e);
+        }
+    }
+    
+    public static Volume getVolumeFrom(DescribeVolumesResponse response) throws FogbowException {
+        if (response != null && !response.volumes().isEmpty()) {
+            return response.volumes().listIterator().next();
+        }
+        throw new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND);
+    }
+    
+    public static DescribeVolumesResponse doDescribeVolumesRequest(Ec2Client client, DescribeVolumesRequest request)
+            throws FogbowException {
+        try {
+            return client.describeVolumes(request);
+        } catch (Exception e) {
+            throw new UnexpectedException(String.format(Messages.Exception.GENERIC_EXCEPTION, e), e);
+        }
+    }
+    
+    public static void createTagsRequest(String resourceId, String key, String value, Ec2Client client)
+            throws FogbowException {
+
+        Tag tag = Tag.builder()
+                .key(key)
+                .value(value)
+                .build();
+
+        CreateTagsRequest request = CreateTagsRequest.builder()
+                .resources(resourceId)
+                .tags(tag)
+                .build();
+        try {
+            client.createTags(request);
+        } catch (Exception e) {
+            throw new UnexpectedException(String.format(Messages.Exception.GENERIC_EXCEPTION, e), e);
+        }
+    }
+
+    public static void doDeleteSecurityGroup(String groupId, Ec2Client client) throws FogbowException {
+        DeleteSecurityGroupRequest request = DeleteSecurityGroupRequest.builder()
+                .groupId(groupId)
+                .build();
+        try {
+            client.deleteSecurityGroup(request);
+        } catch (SdkException e) {
```

Comment:
why are we logging the error in this method and not logging in the others?

Reply:
This logging the error is quite common for crashes while removing a Fogbow resource, this particular method was taken from the Security Rule implementation in the Network and Public IP plugins. I will try to throw this logging the error into the origin plugins.